<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1088</width>
    <height>539</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mesh_Computational_Geometry</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="GLDisplayWidget" name="widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="sizeIncrement">
       <size>
        <width>0</width>
        <height>100</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetFixedSize</enum>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayoutInformations">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="title">
           <string>Mesh creation..</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QPushButton" name="pb2DSquare">
             <property name="text">
              <string>2D Square</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbPyramid">
             <property name="text">
              <string>Pyramid</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbTetrahedron">
             <property name="text">
              <string>Tetrahedron</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbCreateQueen">
             <property name="text">
              <string>Queen</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbOpenFile">
             <property name="text">
              <string>Open OFF file</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
            <underline>false</underline>
           </font>
          </property>
          <property name="title">
           <string>Informations about the mesh</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="labelFaceNb">
             <property name="text">
              <string>Number of faces : </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="labelVertexNb">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Number of vertices : </string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Iterate over the mesh</string>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="label">
               <property name="font">
                <font>
                 <weight>50</weight>
                 <bold>false</bold>
                </font>
               </property>
               <property name="text">
                <string>Start vertex :</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEditStartVertex">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLabel" name="labelCurrFace">
               <property name="text">
                <string>Current face :</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout">
               <item>
                <widget class="QPushButton" name="pbNextFace">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Next face</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pbPrevFace">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Previous face</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLabel" name="labelCurrVertex">
               <property name="text">
                <string>Current neighboor vertex :</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <widget class="QPushButton" name="pbNextVertex">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Next vertex</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pbPrevVertex">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Previous vertex</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>Modification of the mesh</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QGroupBox" name="groupBox_6">
               <property name="title">
                <string>Add a middle point in a triangle</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_11">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_9">
                  <item>
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>Face index :</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLineEdit" name="lineEditFaceIndexSplit">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <widget class="QPushButton" name="pbSplitFaceMiddle">
                  <property name="text">
                   <string>Split face</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QGroupBox" name="groupBox_7">
               <property name="title">
                <string>Add a point by coordinates</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_12">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_2">
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_10">
                    <item>
                     <layout class="QVBoxLayout" name="verticalLayout_13">
                      <item>
                       <widget class="QLabel" name="label_3">
                        <property name="text">
                         <string>X =</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLabel" name="label_4">
                        <property name="text">
                         <string>Y =</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLabel" name="label_5">
                        <property name="text">
                         <string>Z =</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                    <item>
                     <layout class="QVBoxLayout" name="verticalLayout_14">
                      <item>
                       <widget class="QDoubleSpinBox" name="doubleX">
                        <property name="minimum">
                         <double>-99.989999999999995</double>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QDoubleSpinBox" name="doubleY">
                        <property name="minimum">
                         <double>-99.989999999999995</double>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QDoubleSpinBox" name="doubleZ">
                        <property name="minimum">
                         <double>-99.989999999999995</double>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
                <item>
                 <widget class="QPushButton" name="pbAddPoint">
                  <property name="text">
                   <string>Add the point naively</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="pbDelaunayAdd">
                  <property name="text">
                   <string>Add the point with delaunay</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayoutBoutonDroit">
        <property name="spacing">
         <number>2</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>GroupBox</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QPushButton" name="pushButton">
             <property name="text">
              <string>Voronoï diagram</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbWireframe">
             <property name="text">
              <string>Wireframe     ON</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbTestIterators">
             <property name="text">
              <string>Test iterators</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbResetV_F">
             <property name="text">
              <string>Reset vertex/face</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbExit">
             <property name="text">
              <string>Exit</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLDisplayWidget</class>
   <extends>QWidget</extends>
   <header>gldisplaywidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pbExit</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>93</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>action()</slot>
 </slots>
</ui>
